pyi.py

```diff
@@ -743,12 +762,54 @@ def visit_Assign(self, node: ast.Assign) -> None:
         ):
             return self._Y015_error(node)
 
-        # We avoid triggering Y026 for calls and = ... because there are various
-        # unusual cases where assignment to the result of a call is legitimate
-        # in stubs.
-        if not is_special_assignment and not isinstance(
-            assignment, (ast.Ellipsis, ast.Call)
-        ):
+        if not is_special_assignment:
+            self._check_for_type_aliases(node, target_name, assignment)
+
+    def _check_for_type_aliases(
+        self, node: ast.Assign, target_name: str, assignment: ast.expr
+    ) -> None:
+        """
+        Check for assignments that look like they could be type aliases,
+        but aren't annotated with `typing(_extensions).TypeAlias`.
+
+        We avoid triggering Y026 for calls and = ... because there are various
+        unusual cases where assignment to the result of a call is legitimate
+        in stubs (`T = TypeVar("T")`, `List = _Alias()`, etc.).
+
+        Most assignments to names in builtins.py or typing.py will be type aliases,
+        so special-case those. For other `ast.Attribute` and `ast.Name` nodes,
+        avoid triggering Y026 now, as they might be variable aliases rather than
+        type aliases.
+        """
+        if isinstance(assignment, (ast.Ellipsis, ast.Call)):
+            return
+
+        special_cased_modules = {builtins, typing, collections.abc}
+
+        if isinstance(assignment, ast.Attribute):
+            if isinstance(assignment.value, ast.Name):
+                module_name, assignment_name = assignment.value.id, assignment.attr
+                for module in special_cased_modules:
+                    if module_name == module.__name__:
+                        if assignment_name in dir(
+                            module
+                        ) and _is_probably_class_from_module(
+                            assignment_name, module=module
+                        ):
```

Comment:
ew @ black's formatting here.

Reply:
Putting parentheses around the whole thing helps sometimes

Reply:
I tried it and it black'd my parens away :'(